cmd/cstor-volume-mgmt/controller/volume-controller/handler.go

```diff
@@ -74,31 +74,70 @@ func (c *CStorVolumeController) cStorVolumeEventHandler(operation common.QueueOp
 		// CheckValidVolume is to check if volume attributes are correct.
 		err := volume.CheckValidVolume(cStorVolumeGot)
 		if err != nil {
-			return common.CVStatusOffline, err
+			return common.CVStatusError, err
 		}
 
 		err = volume.CreateVolumeTarget(cStorVolumeGot)
 		if err != nil {
-			return common.CVStatusFailed, err
+			return common.CVStatusError, err
 		}
-		break
+
+		return common.CVStatusInit, nil
 
 	case common.QOpModify:
+		// Make changes here to run zrepl command and update the data
 		err := volume.CheckValidVolume(cStorVolumeGot)
 		if err != nil {
 			return common.CVStatusInvalid, err
 		}
 
 		err = volume.CreateVolumeTarget(cStorVolumeGot)
 		if err != nil {
-			return common.CVStatusFailed, err
+			return common.CVStatusError, err
 		}
 		break
 
+	case common.QOpPeriodicSync:
+		err := volume.CheckValidVolume(cStorVolumeGot)
+		if err != nil {
+			return common.CVStatusInvalid, err
+		}
+
+		statuses, err := volume.GetReplicaStatus(cStorVolumeGot)
+		if err != nil {
+			return common.CVStatusError, err
+		}
+
+		glog.Infof("Replica statuses for volume %s are: %v", cStorVolumeGot.Name, statuses)
+
+		healthyReplicas := 0
+		for _, val := range statuses {
+			if val == "HEALTHY" {
+				healthyReplicas++
+			}
+		}
+
+		glog.Infof("Healthy replicas for volume %s are: %v", cStorVolumeGot.Name, healthyReplicas)
+
+		if healthyReplicas == 0 {
+			return common.CVStatusError, nil
```

Comment:
We can probably set this as Init state if healthReplica=0 and totalReplicas=0 -- the target is waiting for replicas to join.